From 30844350dabac95f0c8dd846fd56fbc48830c6d8 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Mon, 15 Feb 2021 12:56:41 +0530
Subject: [PATCH 710/767] octeontx2-af: Remove TOS field from MKEX TX

commit 8d3cfaa94e1cd91e4bae48b475b4a531282d3b20 from
git@git.assembla.com:cavium/WindRiver.linux.git

The TOS field got added to MKEX TX profile when
backporting upstream changes. TOS overlaps with
DMAC field and hence installing rules for TX side
are failing. Hence remove TOS field from TX profile.

Fixes: 9f50807008("octeontx2-af: Backport upstream kernel changes")
Change-Id: I559e7007bbac15f74ad35146ec86492a3b84c1cf
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/46080
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
index c2661cb64bd2..4059dd7c878e 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
@@ -16220,8 +16220,6 @@ static struct npc_mcam_kex npc_mkex_default = {
 			[NPC_LT_LC_IP] = {
 				/* SIP+DIP: 8 bytes, KW2[63:0] */
 				KEX_LD_CFG(0x07, 0xc, 0x1, 0x0, 0x10),
-				/* TOS: 1 byte, KW1[63:56] */
-				KEX_LD_CFG(0x0, 0x1, 0x1, 0x0, 0xf),
 			},
 			/* Layer C: IPv6 */
 			[NPC_LT_LC_IP6] = {
-- 
2.31.1

